iModelCore/BeSQLite/BeSQLite.cpp

```diff
@@ -5361,6 +5361,19 @@ Utf8CP BeSQLiteLib::GetErrorName(DbResult code) {
         case BE_SQLITE_ERROR_CouldNotAcquireLocksOrCodes: return "BE_SQLITE_ERROR_CouldNotAcquireLocksOrCodes";
         case BE_SQLITE_ERROR_SchemaUpgradeRecommended:    return "BE_SQLITE_ERROR_SchemaUpgradeRecommended";
         case BE_SQLITE_ERROR_NOTOPEN:                     return "BE_SQLITE_ERROR_NOTOPEN";
+        case BE_SQLITE_ERROR_SchemaNotFound:              return "BE_SQLITE_ERROR_SchemaNotFound";
+        case BE_SQLITE_ERROR_SchemaReadFailed:            return "BE_SQLITE_ERROR_SchemaReadFailed";
+        case BE_SQLITE_ERROR_SchemaIsDynamic:             return "BE_SQLITE_ERROR_SchemaIsDynamic";
+        case BE_SQLITE_ERROR_SchemaDomainNamesMismatched: return "BE_SQLITE_ERROR_SchemaDomainNamesMismatched";
+        case BE_SQLITE_ERROR_DbIsReadonly:                return "BE_SQLITE_ERROR_DbIsReadonly";
+        case BE_SQLITE_ERROR_FailedToParseXml:            return "BE_SQLITE_ERROR_FailedToParseXml";
+        case BE_SQLITE_ERROR_InvalidECSchemaXml:          return "BE_SQLITE_ERROR_InvalidECSchemaXml";
+        case BE_SQLITE_ERROR_ReferencedSchemaNotFound:    return "BE_SQLITE_ERROR_ReferencedSchemaNotFound";
+        case BE_SQLITE_ERROR_DuplicateSchema:             return "BE_SQLITE_ERROR_DuplicateSchema";
+        case BE_SQLITE_ERROR_DuplicateTypeName:           return "BE_SQLITE_ERROR_DuplicateTypeName";
+        case BE_SQLITE_ERROR_InvalidPrimitiveType:        return "BE_SQLITE_ERROR_InvalidPrimitiveType";
+        case BE_SQLITE_ERROR_HasReferenceCycle:           return "BE_SQLITE_ERROR_HasReferenceCycle";
+        case BE_SQLITE_ERROR_PruneItem:                   return "BE_SQLITE_ERROR_PruneItem";

```

Comment:
Why are EC schema errors being surfaced as SQLite errors?

Reply:
Well, DbResult is supposed to represent sqlite errors, not EC errors. But I see we have previously already polluted it with "iModelPlatform custom errors" like SchemaImporetFailed. @khanaffan @ColinKerr?

Reply:
I've removed the newly added error codes. Instead, APIs will now throw exceptions with the error number upon failure.

Reply:
I just had a look at the changes. The code looks good, but switching the return value from something to void, and starting to throw exceptions from a place that previously didn't throw exceptions both sounds like breaking changes. Callers will need to be adjusted, no? Are we good to make that kind of change?

Reply:
These changes are safe to make because only our internal functions use these APIs and are not exposed to the users. There will be no impact on user-facing methods.

Reply:
Sounds good, thanks.